src/rpc/CredentialHelpers.cpp

```diff
@@ -53,6 +53,7 @@
 namespace rpc::credentials {
 
 bool
+// NOLINTNEXTLINE(misc-use-internal-linkage)
```

Comment:
Probably we are doing something wrong with this check or it generates too many false positives.

Reply:
I think it's just the way we have global functions that become unused (within the internal library it compiles for) if we mark them `static`. So the solution is to not have these global freestanding functions. This is not something I wanted to change within this PR. I think this can be addressed separately later. And this check enabled now means that we can no longer add those global functions and shall find a suitable workaround next time we need them. I am sure a simple wrapper class with static functions would suffice (e.g. `NFTUtils`).